commands/reqlog.go

```diff
@@ -38,13 +38,13 @@ func (rl *ReqLog) AddEntry(rle *ReqLogEntry) {
 	rl.lock.Lock()
 	defer rl.lock.Unlock()
 
-	rle.ID = rl.nextID
-	rl.nextID++
-	rl.Requests = append(rl.Requests, rle)
-
 	if rle == nil || !rle.Active {
 		rl.maybeCleanup()
 	}
+
+	rle.ID = rl.nextID
+	rl.nextID++
+	rl.Requests = append(rl.Requests, rle)
 }
```

Comment:
Being consistent with the rest of the methods, I would say that removing the nil check is ok.

Reply:
Seems reasonable to me, it's certainly no worse than what we currently have. We may want to revisit what's going on with the ReqLog later anyway because I see that the code is duplicated here and in https://github.com/ipfs/go-ipfs-cmds/blob/master/reqlog.go and probably one of these should be deleted.

However, we can do this further down the road